actionpack/lib/action_controller/metal/params_wrapper.rb

```diff
@@ -119,6 +119,14 @@ def include
                 end
               end
 
+              if m.respond_to?(:reflections) && m.reflections.keys.any?
+                self.include += m.reflections.select do |key, reflection|
+                  reflection.class == ActiveRecord::Reflection::HasAndBelongsToManyReflection
```

Comment:
Why `has_and_belongs_to_many` is special? Should not `has_many through:` also be included here?